notification/src/handler/notification/notification.handler.js

```diff
@@ -154,6 +159,34 @@ function calculateUpdateActionsForPayment(payment, notification) {
   return updateActions
 }
 
+/**
+ * Compares transaction states
+ * @param currentState state of the transaction from the CT platform
+ * @param newState state of the transaction from the Adyen notification
+ * @return number 1 if newState can appear after currentState
+ * -1 if newState cannot appear after currentState
+ * 0 if newState is the same as currentState
+ * @throws Error when newState and/or currentState is a wrong transaction state
+ * */
+function compareTransactionStates(currentState, newState) {
+  const transactionStateFlow = {
+    Initial: 0,
+    Pending: 1,
+    Success: 2,
+    Failure: 2,
+  }
+  if (
+    !transactionStateFlow.hasOwnProperty(currentState) ||
+    !transactionStateFlow.hasOwnProperty(newState)
+  )
+    throw Error(
+      'Wrong transaction state passed. ' +
+        `currentState: ${currentState}, newState: ${newState}`
+    )
+
+  return transactionStateFlow[newState] - transactionStateFlow[currentState]
+}
+
 function getAddInterfaceInteractionUpdateAction(notification) {
   // strip away sensitive data
   delete notification.additionalData
```

Comment:
maybe we should not remove all data as metadata might be good to have for tracability also modification.action looks like good to have ?

Reply:
We could, but it means we would need to check what data could be passed in `additionalData` and what they can contain. And we need to verify it. And it could have lots of possible fields inside: https://docs.adyen.com/development-resources/webhooks/additional-settings.